share/gpodder/extensions/youtube-dl.py

```diff
@@ -285,6 +285,10 @@ def __init__(self, gpodder_config, my_config, force=False):
             self._ydl_opts['verbose'] = True
         else:
             self._ydl_opts['quiet'] = True
+        # Don't create downloaders for URLs supported by these youtube-dl extractors
+        self.ie_blacklist = ["Generic"]
+        # Cache URL regexes from youtube-dl matches here, seed with youtube regex
+        self.regex_cache = [re.compile(r'https://www.youtube.com/watch\?v=.+')]
```

Comment:
instead of maintaining the regex_cache, can you load all regexps and combine them at startup?
(untested)
```
all_patterns = [(r'https://www.youtube.com/watch\?v=.+')] + [ie._VALID_URL_RE.pattern for ie in ydl._ies if ie.ie_key() not in self.ie_blacklist]
self.regex_cache = re.compile('((?:' + ')|(?:'.join(all_patterns) + '))')
```
Please try it out and let us know if it's more efficient than managing the cache regex list.

Reply:
Thanks for taking the time to try it out! I haven't this function readily available, unfortunately